builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -642,8 +673,28 @@ func resourceVSphereVirtualMachineCreate(d *schema.ResourceData, meta interface{
 		for i, v := range vL.([]interface{}) {
 			disk := v.(map[string]interface{})
 			if i == 0 {
-				if v, ok := disk["template"].(string); ok && v != "" {
-					vm.template = v
+				if diskClone, ok := disk["clone"].(map[string]interface{}); ok && len(diskClone) > 0 {
```

Comment:
Actually, it's not really because of the `disk := ...` line. It's because the schema will always provide it even if it's not set, so it will always be accessible via the `disk` variable.